Pennsieve/pennsieve-app-2
# Reset password cognito- cu-p30z01

# Description

The purpose of this PR is to add reset password functionality for Cognito. In addition, I moved the Amplify config setup to `main.js` as [noted in the Amplify docs](https://docs.amplify.aws/lib/auth/getting-started/q/platform/js#configure-your-application).

## Clickup Ticket

[p30z01](https://app.clickup.com/t/p30z01)


## Type of change


# How Has This Been Tested?

- On the login page, click on "Forgot your password?"
- Enter your email, and you should see a "Reset link sent." screen
- Check your email for a verification code
- Copy verification code and go to the following email, replacing the variables in the URL with your email and verification code
  - http://localhost:3000/password?verificationCode=INSERT_CODE&email=INSERT_EMAIL
  - Please note, that if you have `+` in your email, you should encode this query parameter like the email will when sent, so `email=cameron+foo@blackfynn.com` should become `email=cameron%2Bfoo@blackfynn` in the URL.
- You should be taken to the reset page with the email and code input filled out, and autofocused on the password field
- Enter a new password, and if successful, you should see a success toast and be taken back to the login page.
- Login with your new password (should succeed in the network for now, until the login PR is done)

# Checklist:
